pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -667,22 +668,58 @@ func getVirtualDiskUUID(newDevice types.BaseVirtualDevice) (string, error) {
 	vd := newDevice.GetVirtualDevice()
 
 	if b, ok := vd.Backing.(*types.VirtualDiskFlatVer2BackingInfo); ok {
-		uuidWithNoHypens := strings.Replace(b.Uuid, "-", "", -1)
-		return strings.ToLower(uuidWithNoHypens), nil
+		uuid := formatVirtualDiskUUID(b.Uuid)
+		return uuid, nil
 	}
 	return "", ErrNoDiskUUIDFound
 }
 
-func getVirtualDiskID(volPath string, vmDevices object.VirtualDeviceList) (string, error) {
+func formatVirtualDiskUUID(uuid string) string {
+
+	uuidwithNoSpace := strings.Replace(uuid, " ", "", -1)
+	uuidWithNoHypens := strings.Replace(uuidwithNoSpace, "-", "", -1)
+	uuid = strings.ToLower(uuidWithNoHypens)
+
+	return uuid
+}
+
+//Gets virtual disk uuid by datastore path. Here, datastore path can be both containing vsan object name or uuid.
```

Comment:
whats a vsan object name, is it a datastore vmdk path? does this mean volPath argument can be a uuid as well?